### ISSUE TITLE
Move authorisation spa to typescript

### ISSUE THREAD

**Issue by kritika-singh3** [gocd/gocd#5552]
##### Summary
current spa is implemented using mithril. This issue is to move it to typescript. 

##### Todo
- [ ] no api changes required. Just client-side implementation needs to change to typescript

**Comment by kritika-singh3** [comment]
We noticed small change in the position of the buttons on Modal. I guess we are following this across all new SPA. This is just to let everyone know.

#### Old SPA
<img width="1222" alt="screen shot 2018-12-14 at 12 23 04 pm" src="https://user-images.githubusercontent.com/41165891/49988170-d5fde480-ff9b-11e8-9aef-420b11eb7db9.png">

#### New SPA
<img width="1103" alt="screen shot 2018-12-14 at 12 22 49 pm" src="https://user-images.githubusercontent.com/41165891/49988178-dc8c5c00-ff9b-11e8-9549-543d2a8db7c7.png">


/cc @naveenbhaskar

**Comment by naveenbhaskar** [comment]
yes.  all user interactions  in the user flow are in the right side.
